pkg/operator/resource/resourceapply/monitoring.go

```diff
@@ -139,6 +139,10 @@ func ApplyUnstructuredResourceImproved(
 	// Check if the metadata objects differ. This only checks for selective fields (excluding the resource version, among others).
 	didMetadataModify := ptr.To(false)
 	resourcemerge.EnsureObjectMeta(didMetadataModify, &existingObjectMetaTyped, requiredObjectMetaTyped)
+	existingCopy.Object["metadata"], err = runtime.DefaultUnstructuredConverter.ToUnstructured(&existingObjectMetaTyped)
+	if err != nil {
+		return nil, false, err
+	}
```

Comment:
Note that this is a [breaking change](https://github.com/rexagod/cluster-monitoring-operator/blob/70fc3336873b4cc508f14e484b844ee601720f98/pkg/client/client.go#L671-L704), as earlier the method was only used to detect if the metadata fields differed, but not actually mutate them.

Reply:
Previously the metadata from the `required` obj were merged to the `existing` obj but  the `required` obj was stored in the database. I thin this behaviour was unexpected, am I right ?